### ISSUE TITLE
QoS seems not to be working

### ISSUE THREAD

**Issue by sanderkukk** [oci-labs/node-opendds#34]
Hello,

Im using latest node-opendds master `94f20e8152a5daaa3c3c893606c85cad5c2c8d5c` and latest OpenDDS release (3.19.0).
It seems to me that setting QoS is not working as expected. All new generated messages are received but not old messages.
I am setting "TRANSIENT_LOCAL_DURABILITY_QOS" also tried "RELIABLE_RELIABILITY_QOS" to no effect.
Can confirm this is working with latest node-opendds master and openDDS with commit `2550e2654208f1ce382b71ad8a0075b7f2152a8f`

Currently trying to reproduce the issue with the built-in test. Do you have any quick debug pointers where should I look? Will for example I see the issue with Wireshark?

**Comment by simpsont-oci** [comment]
The test should already be setting `RELIABLE_RELIABILITY_QOS`. Reliability is the default for datawriters, but needs to be explicitly set for datareaders (where the default according to the spec is best effort). Looking at the code, I see that's true for test_subscriber.js, but not for test_subscriber.cpp, which is a bit confusing. I'll submit a PR to update the C++ test subscriber but it looks like it hasn't had any troubles getting best effort messages in the CI environment.

Could you give a little more info about your test setup? What languages are you using for your publishing and subscribing applications? Are you setting transient local durability for both the datawriter and datareader, or just the datawriter? What are your HISTORY and RESOURCE_LIMITS QoS settings for both?

**Comment by simpsont-oci** [comment]
https://github.com/oci-labs/node-opendds/pull/35

**Comment by sanderkukk** [comment]
Thanks again for the quick answer.
I diagnosed it more and my assessment of the problem being in QoS was wrong - well partially.

>  Are you setting transient local durability for both the datawriter and datareader, or just the datawriter? What are your HISTORY and RESOURCE_LIMITS QoS settings for both?
This gave me a better direction towards the issue. I was setting durability for both. What did I not set was `reliability` and `history`. I only set `durability` - well I have had always but the jump to the latest openDDS broke that. Maybe before, it defaulted to some / different values than now.

Now I'm seeing everything working as expectedly. Thanks!
Marking this as closed!
